pkg/api/store/service/service.go

```diff
@@ -42,3 +48,32 @@ func formatData(data map[string]interface{}) {
 	ports = append(ports, m)
 	data["ports"] = ports
 }
+
+func (p *Store) validateNonSpecialIP(schema *types.Schema, data map[string]interface{}) error {
+	if schema.ID == "dnsRecord" {
+		switch ips := data["ipAddresses"].(type) {
+		case []interface{}:
+			for _, ip := range ips {
```

Comment:
you can just do the direct conversion, no need to use switch as there is only one possible value